Fix patches 0013 and 0014 for open issues

giuliobenetti opened the pull request:
@mripard 
Fix 0014 fixes https://github.com/mripard/sunxi-mali/issues/12
Fix 0013 fixes https://github.com/mripard/sunxi-mali/issues/11

I don't know if I've written correctly here.
Is there a standard way when PR multiple commits?

giuliobenetti committed:
Fix 0014-mali-Make-devfreq-optional.patch

0014-mali-Make-devfreq-optional.patch can't be applied

This is due to wrong path specified inside the patch
that doesn't take into account mali version to be patched

Removed r6p2 from every path inside patch

Signed-off-by: Giulio Benetti <giulio.benetti@micronovasrl.com>

sergey-suloev reviewed (commented):
Hey Giulio,
I found that there is "setup_timer" wrapper around "__setup_timer" function. Better use it, I think.

giuliobenetti commented:
Hi @sergey-suloev,
yes you're right, it would be more elegant and setup_timer wrapper is supported since v4.4.
If Maxime applies my Pull Request, why don't you try to add this little modification?

sergey-suloev commented:
@giuliobenetti  I forked the branch and can modify what I need. Fast & furious.

giuliobenetti commented:
Great, now let's wait Maxime to understand how to proceed.

mripard commented on patches/0013-mali-support-building-against-4.15.patch:
```diff
@@ -108,7 +108,7 @@ Signed-off-by: John Stultz <john.stultz@linaro.org>
 -	tim->timer.data = (unsigned long)data;
 -	tim->timer.function = (timer_timeout_function_t)callback;
-+	timer_setup(&tim->timer, callback, 0);
++	__setup_timer(&tim->timer, callback, &tim->timer, 0);
```
Please juste add the definition of timer_setup for the older kernels.

giuliobenetti replied:
Do you mean to place #if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 14, 0) and use different apis?

giuliobenetti replied:
Or define timer_setup api locally and generate 0015 patch?

sergey-suloev replied:
something like this: 

#if kernel < ver
#define timer_setup(_a, _b, _c) setup_timer(_a, b, blabla, _c)
#endif

giuliobenetti committed:
Add patch for building with kernel version < 4.14

In 0013-mali-support-building-against-4.15.patch
has been used timer_setup api.
This api has been introduced only in 4.14.0 version,
so compiling with version < 4.14.0 fails.

Added function locally to mali_osk_timers.c compiled
if version < 4.14.0

Signed-off-by: Giulio Benetti <giulio.benetti@micronovasrl.com>
